packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -47,21 +55,66 @@ const GridFilterFormRoot = styled('div', {
   overridesResolver: (props, styles) => styles.filterForm,
 })(({ theme }) => ({
   display: 'flex',
-  justifyContent: 'space-around',
   padding: theme.spacing(1),
 }));
 
+const getLinkOperatorLocaleKey = (linkOperator) => {
+  switch (linkOperator) {
+    case GridLinkOperator.And:
+      return 'filterPanelOperatorAnd';
+    case GridLinkOperator.Or:
+      return 'filterPanelOperatorOr';
+    default:
+      throw new Error('MUI: Invalid `linkOperator` property in the `GridFilterPanel`.');
+  }
+};
+
+const getColumnLabel = (col) => col.headerName || col.field;
+
+const useColumnSorting = (filterableColumns, columnsSort) => {
+  const [sortedColumns, setSortedColumns] = React.useState(filterableColumns);
+
+  React.useEffect(() => {
+    switch (columnsSort) {
+      case 'asc':
+        setSortedColumns(
+          filterableColumns.sort((a, b) => (getColumnLabel(a) < getColumnLabel(b) ? -1 : 1)),
```

Comment:
Comparing `'ã' < 'b'` is false. Prefer the [`Intl.Collator`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator) API which correctly sorts accented words.

https://github.com/mui-org/material-ui-x/blob/99d0d24cf638defc2e76933892a9f15d40fcf6e8/packages/grid/_modules_/grid/utils/sortingUtils.ts#L44